### PR TITLE
Grouped list box

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

Updates ListBox to handle "groups"

Screen Shot 2023-02-14 at 8 55 09 AM

- Updates ListBox styling for groups - Heading + List styles when focused or hovered
- Add scrollTo to scroll the List when using the keyboard and targeting an element outside of the modal height
- Updates wording to use "Block"

# Test instructions | Instructions pour tester la modification

- Test updates listed above to the Add Element / Add Block dialog

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

**Comment by timarney** [comment]
@anikbrazeau 

FYI - there are some updated FR strings in this PR

https://github.com/cds-snc/platform-forms-client/pull/1615/files#diff-a6e486b021382fa7935492752ec9890787007f55c22fdc5c300ec95e11355a7bR4

Feel free to make a commit here if you would like to update in this PR.

File: public/static/locales/fr/form-builder.json [hunk 1]
```diff
@@ -1,16 +1,19 @@
 {
   "addElement": "Ajouter un élément",
   "addElementDialog": {
-    "addButton": "Sélectionner un élément",
-    "questionElement": "Élément de question",
+    "layoutBlocks": "[FR] Layout blocks",
+    "inputBlocks": "[FR] Input blocks",
+    "advancedBlocks": "[FR] Advanced blocks",
+    "addButton": "[FR] Select block",
+    "questionElement": "[FR] Blocks",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:8]:
```suggestion
    "layoutBlocks": "Blocs de mise en page",
    "inputBlocks": "Blocs de saisie",
    "advancedBlocks": "Blocs avancés",
    "addButton": "Sélectionner un bloc",
    "questionElement": "Blocs",
```

File: public/static/locales/fr/form-builder.json [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
     "listItem": "Élément de liste ",
     "checkboxItem": "Élément de case à cocher",
     "example": "exemple",
     "answer": "Réponse...",
     "yes": "Oui",
     "no": "Non",
     "sometimes": "Parfois",
-    "exampleElement": "élément d'exemple",
+    "exampleElement": "[FR] example block",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:16]:
```suggestion
    "exampleElement": "exemple d'un bloc",
```

File: public/static/locales/fr/form-builder.json [hunk 3]
```diff
@@ -52,7 +55,7 @@
       "description": "Description du numéro"
     },
     "dynamicRow": {
-      "title": "[FR] Repeating question",
+      "title": "[FR] Repeating set",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:58]:
```suggestion
      "title": "Série répétitive de questions",
```

File: public/static/locales/fr/form-builder.json [hunk 4]
```diff
@@ -334,7 +337,7 @@
   "translateTitle": "Saisir la traduction",
   "unlockPublishing": "Soumettre",
   "unlockPublishingDescription": "Pour déverrouiller la publication, fournissez les coordonnées de la personne de qui vous relevez. ",
-  "dyanamicRow": "[FR] Repeating question set",
+  "dyanamicRow": "[FR] Repeating set",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:340]:
```suggestion
  "dyanamicRow": "Série répétitive de questions",
```

File: public/static/locales/fr/form-builder.json [hunk 5]
```diff
@@ -334,7 +337,7 @@
   "translateTitle": "Saisir la traduction",
   "unlockPublishing": "Soumettre",
   "unlockPublishingDescription": "Pour déverrouiller la publication, fournissez les coordonnées de la personne de qui vous relevez. ",
-  "dyanamicRow": "[FR] Repeating question set",
+  "dyanamicRow": "Série répétitive de questions",
   "addToSet": "[FR] Add to set",
   "removeFromSet": "[FR] Remove from set",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:342]:
```suggestion
  "addToSet": "Ajouter à la série",
  "removeFromSet": "Enlever de la série",
```

**Comment by anikbrazeau** [comment]
[Review GC Forms builder microcopy for bilingual parity, consistency and clarity improvements](https://trello.com/c/h2SiWmZK/5-review-gc-forms-builder-microcopy-for-bilingual-parity-consistency-and-clarity-improvements)

**Review by dsamojlenko — APPROVED** [review]

**Pull request merged**